File: healthcare-api/models/FoodLog.js
// healthcare-api/models/FoodLog.js
const mongoose = require('mongoose');

const foodLogSchema = new mongoose.Schema(
  {
    userId:   { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    mealType: { type: String, enum: ['breakfast', 'lunch', 'dinner', 'snack'], required: true },
    foodName: { type: String, required: true, trim: true },
    calories: { type: Number, required: true, min: 0 },
    protein:  { type: Number, default: 0, min: 0 },
    carbs:    { type: Number, default: 0, min: 0 },
    fats:     { type: Number, default: 0, min: 0 },
    fiber:    { type: Number, default: 0, min: 0 },
    calcium:  { type: Number, default: 0 },
    iron:     { type: Number, default: 0 },
    magnesium:{ type: Number, default: 0 },
    date:     { type: String, required: true }, // YYYY-MM-DD
    time:     { type: String, default: () => new Date().toLocaleTimeString('en-US', { hour: '2-digit', minute: '2-digit' }) },
  },
  { timestamps: true }
);

// Compound index for fast daily queries
foodLogSchema.index({ userId: 1, date: 1 });

module.exports = mongoose.model('FoodLog', foodLogSchema);

File: healthcare-fastapi/models/food_log.py
# healthcare-fastapi/models/food_log.py
import uuid
from datetime import datetime
from sqlalchemy import DateTime, Float, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column
from database.db import Base


class FoodLog(Base):
    __tablename__ = "food_logs"

    id: Mapped[str] = mapped_column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id: Mapped[str] = mapped_column(String, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)

    date: Mapped[str] = mapped_column(String(10), nullable=False, index=True)  # YYYY-MM-DD
    meal_type: Mapped[str] = mapped_column(String(20), nullable=False)          # breakfast/lunch/snack/dinner
    food_name: Mapped[str] = mapped_column(String(200), nullable=False)
    time: Mapped[str] = mapped_column(String(20), nullable=True)

    calories: Mapped[float] = mapped_column(Float, default=0)
    protein: Mapped[float] = mapped_column(Float, default=0)
    carbs: Mapped[float] = mapped_column(Float, default=0)
    fats: Mapped[float] = mapped_column(Float, default=0)
    fiber: Mapped[float] = mapped_column(Float, default=0)

    # Premium micronutrients
    calcium: Mapped[float] = mapped_column(Float, nullable=True)
    iron: Mapped[float] = mapped_column(Float, nullable=True)
    magnesium: Mapped[float] = mapped_column(Float, nullable=True)

    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
